membership/src/main/scala/zio/membership/hyparview/Views.scala

```diff
@@ -185,10 +179,24 @@ private[hyparview] object Views {
         override def addShuffledNodes(sentOriginally: Set[T], replied: Set[T]) =
           for {
             _         <- passiveView0.removeIf(sentOriginally.contains)
+            size      <- passiveViewSize
+            _         <- dropNFromPassive(replied.size - (passiveViewCapacity - size))
             _         <- addAllToPassiveView(replied.toList)
             remaining <- passiveViewSize.map(passiveViewCapacity - _)
             _         <- addAllToPassiveView(sentOriginally.take(remaining).toList)
           } yield ()
+
+        private def dropNFromPassive(n: Int): STM[Nothing, Unit] = n match {
+          case x if x <= 0 => STM.unit
+          case _           => dropOneFromPassive *> dropNFromPassive(n - 1)
+        }
+
+        private val dropOneFromPassive =
+          for {
+            list    <- passiveView0.toList
+            dropped <- tRandom.selectOne(list)
+            _       <- STM.foreach(dropped)(passiveView0.delete(_))
```

Comment:
Make a note (or open an issue) to replace these with `STM.foreach_` as soon as we have ZIO upgrade.